### ISSUE TITLE
Update GitHub Actions versions in dev.yml and prod.yml

### ISSUE THREAD

**Issue by mjbear** [aristanetworks/ci-workshops-avd#87]
Hello Arista Engineering people,

I still had some time on the temporary ATD lab environment and couldn't resist working through the CI/CD section!
In going through that section I discovered that GitHub has made some changes to supported versions of Node.js so some versions of GitHub Actions need adjusted.

**Please test that you too have issues with the GitHub Actions at the pinned versions the lab uses.**
:arrow_right: I'd be glad to submit a PR after receiving your blessing. Thank you!

```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/setup-python@v3, actions/cache@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

In `dev.yml` I made the following changes:
```
uses: actions/checkout@v3 >> v4

uses: actions/setup-python@v3 >> v5

uses: dorny/paths-filter@v2 >> v3
(in two locations)
```

In `prod.yml` I made the same changes as dev:
```
uses: actions/checkout@v3 >> v4

uses: actions/setup-python@v3 >> v5

uses: dorny/paths-filter@v2 >> v3
(in two locations)
```

Note: Since `dev.yml` file output exists in the workshop, there is an [issue over there for discussion](https://github.com/aristanetworks/avd-workshops/issues/151) prior to me submitting a PR. Thank you!

**Comment by mjbear** [comment]
PR #89 will solve this

**Comment by JulioPDX** [comment]
Fixed in #89
